### PR TITLE
[BugFix] Chunked json for txn stream load

### LINKED ISSUE #5515: chunked json payload parsing failed
### Steps to reproduce the behavior (Required)
1. do stream load with json file of size 100MB

### Expected behavior (Required)
stream load success.
### Real behavior (Required)
stream load failed silently.
```
W0425 16:49:26.979848 1331570 json_scanner.cpp:726] illegal json started with [n]
```
### StarRocks version (Required)
main, branch-2.2

### PR DESCRIPTION
**Pull request by rickif**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #5515

## Problem Summary(Required) ：

This PR adds the support of chunked json for transaction stream load.
The total size of transaction would not be limited, but the data size of the single write in the transaction would be limited to 4GB.

### CODE REVIEW

**Comment by rickif** [comment]
This PR is abort to be merged to branch-2.3, since there's something wrong with the planner of the main branch. So I test the case on the branch-2.3. After this PR is merged, I would make cherry-picks.

**Comment by rickif** [comment]
run starrocks_be_unittest

**Review by chaoyli — DISMISSED** [review]

File: be/src/http/action/transaction_stream_load.cpp [hunk 1]
```diff
@@ -410,23 +419,53 @@ void TransactionStreamLoadAction::on_chunk_data(HttpRequest* req) {
     auto evbuf = evhttp_request_get_input_buffer(ev_req);
 
     int64_t start_read_data_time = MonotonicNanos();
-    while (evbuffer_get_length(evbuf) > 0) {
-        ByteBufferPtr bb = ByteBuffer::allocate(4096);
+
+    size_t len = 0;
+    while ((len = evbuffer_get_length(evbuf)) > 0) {
```
Comment (chaoyli) [inline be/src/http/action/transaction_stream_load.cpp:424]: It's better to add a test case to test the large json.

Comment (rickif) [reply be/src/http/action/transaction_stream_load.cpp:424]: I'll add them to the daily test.

File: be/src/runtime/stream_load/stream_load_context.h [hunk 2]
```diff
@@ -199,6 +199,8 @@ class StreamLoadContext {
 
     // buffer for reading data from ev_buffer
     static constexpr size_t kDefaultBufferSize = 64 * 1024;
+    // max buffer size for JSON format is 4GB.
+    static constexpr int64_t kJSONMaxBufferSize = 4 * 1024 * 1024 * 1024;
```
[the remark this answers is not in the document]
Comment (chaoyli) [reply be/src/runtime/stream_load/stream_load_context.h]: If the max JSON is 4GB, you can keep it not changed.

**Review by chaoyli — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by rickif** [comment]
@mergifyio backport main

**Pull request merged**